pkg/network/usm/kafka_monitor_test.go

```diff
@@ -699,13 +700,14 @@ func makeRecordBatch(records ...kmsg.Record) kmsg.RecordBatch {
 	return recordBatch
 }
 
-func makeFetchResponseTopicPartition(recordBatches ...kmsg.RecordBatch) kmsg.FetchResponseTopicPartition {
+func makeFetchResponseTopicPartition(errorCode int16, recordBatches ...kmsg.RecordBatch) kmsg.FetchResponseTopicPartition {
 	respParition := kmsg.NewFetchResponseTopicPartition()
 
 	for _, recordBatch := range recordBatches {
 		respParition.RecordBatches = recordBatch.AppendTo(respParition.RecordBatches)
 	}
 
+	respParition.ErrorCode = errorCode
```

Comment:
Perhaps it would be simpler to just do this assignment in the caller which needs it rather than passing it in as an argument to this function, since only a minority need it (similar to how `AbortedTransactions` is handled)?

Reply:
Good point